### ISSUE TITLE
Docker contacts 3rd party tracking service Wootric without approval or notice

### ISSUE THREAD

**Issue by driverdan** [docker/for-mac#2122]
### Expected behavior
Docker's EULA and privacy policy make no mention of 3rd party tracking in their client side software. The expected behavior is no 3rd party tracking services embedded in Docker's software.

### Actual behavior
The Docker CE for Mac installer for 17.09.0-ce-mac33 (19543) makes a call to `api.wootric.com` without asking for permission from the user. This appears to be a violation of Docker's privacy policy.

I did not MiTM the call so I'm unsure what data is being communicated.

### Information

  - Wootric is a user tracking and surveying service not associated with Docker Inc.

### Steps to reproduce the behavior

  1. Upgrade existing installation of Docker CE for Mac to 17.09.0-ce-mac33 (19543)
  2. At the end of the upgrade the installer will make a call to `api.wootric.com`

**Comment by rask** [comment]
I wonder if this is the case on other platforms as well.

**Comment by d2s** [comment]
Same issue here. Fortunately, people can block the outbound requests with an outbound firewalls, like [Little Snitch](https://obdev.at/products/littlesnitch/index.html) etc.

Would be a good thing to get the ability to disable the analytics directly, instead of having to block the communication manually with a firewall.

**Comment by Manouchehri** [comment]
It's also now making a request to `segment.io`.

**Comment by Manouchehri** [comment]
Did a quick static check, wootric.com seems to track version numbers.

![image](https://user-images.githubusercontent.com/7232674/33237996-31bb90ac-d251-11e7-81b9-55ce117a35ce.png)

URLs: `https://api.wootric.com/v1/end_users/`, `https://api.wootric.com/oauth/token`, `https://api.wootric.com/v1/end_users?access_token=`

`segment.io` looks like usage analytics.

![image](https://user-images.githubusercontent.com/7232674/33238002-5ac12110-d251-11e7-9ec2-efa31f9b8669.png)

URLs: `https://api.segment.io/v1/identify`, `https://api.segment.io/v1/track`

**Comment by shaiguitar** [comment]
FWIW, when denying the request to wootric, the docker daemon crashes.

**Comment by YRM64** [comment]
Tagged #2122 (Github), and opened up by driverdan approximately 7 months ago.

**Comment by driverdan** [comment]
I believe it now asks for permission thanks to GDPR. Can anyone confirm?

**Comment by Manouchehri** [comment]
![image](https://user-images.githubusercontent.com/7232674/43995374-83def41a-9d61-11e8-9ae0-fe9001514170.png)

**Comment by joelanders** [comment]
Just did an install; LittleSnitch told me about this tracking call; no GDPR consent dialog first.

**Comment by doktornotor** [comment]
> I believe it now asks for permission thanks to GDPR. Can anyone confirm?

No, I cannot confirm this, plus, identical issue exists on Windows.

**Comment by eigenhombre** [comment]
This issue still occurs.

**Comment by cpoetter** [comment]
Issue still persists. Although `Send usage statistics` is unchecked, I see calls to api.setment.io and api.wootric.com.

**Comment by doktornotor** [comment]
Listen there, Docker guys: you are in breach of GDPR. Either fix it quick, or you'll get into trouble. Your radio silence is not acceptable.

**Comment by mpptx** [comment]
Bump! Just discovered that, I was thinking that they were some of docker-owned domains, but they're clearly not.

**Comment by cassianoleal** [comment]
Same here. New Docker install on macOS, contacts wootric before asking for consent. Little Snitch tipped me to the malicious request, which then I blocked.

_After_ that, I searched and found this issue and found the tick box in the preferences.

**Comment by christianflintrup** [comment]
Same here, Little Snitch caught the malicious requests when launching Docker for the first time.

**Comment by nebuk89** [comment]
Hey, thanks for you questions. Docker's desktop software is a download that each user has the opportunity to determine whether to download or not. The preference settings associated with the software allow each user to determine their own privacy settings. The only telemetry which Docker otherwise tracks is an anonymized identification number which allows Docker to track the number and version of instances  running . That is consistent with Docker's privacy policy and GDPR.

**Comment by cassianoleal** [comment]
> The preference settings associated with the software allow each user to determine their own privacy settings.

This is a blatant lie. If I can only disable it *after* opening the software, at which point it has already sent my information over, then it does *not* allow me to determine my privacy settings.

**Comment by mpptx** [comment]
There should be some kind of modal where users can make their decision about telemetry. Users are probably even unaware that Docker shares their data with third-party companies and the only way to get to know it is to open settings menu - what probably does not happen very often. I would really appreciate if you consider making that more clear - cause after all your company's products are doing a very good job.

--

Pozdrawiam / with regards

Michał
Sent from my iPhone

On 16 Mar 2020, at 18:01, Cassiano Leal <notifications@github.com> wrote:

﻿

The preference settings associated with the software allow each user to determine their own privacy settings.

This is a blatant lie. If I can only disable it after opening the software, at which point it has already sent my information over, then it does not allow me to determine my privacy settings.

**Comment by nebuk89** [comment]
Sorry for the slow follow up. 

We have the 'welcome whale' that pops up and welcomes you and informs you about the opt in on data collection (https://docs.docker.com/docker-for-windows/install/#start-docker-desktop). 
We could add some logic around this that until this is dismissed or for N time after it is shown we don't send the app lifecycle information etc?  
This pop up is going to be larger going forward so hopefully this will make things a bit clearer as well. 

Or we could go all the way to using a PKG for Mac but I don't think this is something that gives any more value (unless there are other needs?) 

And just to be clear- we used third party tools we do not share the data with them to analyze or make use of :)

**Comment by davidkuridza** [comment]
Same here, the `Send usage statistics` is disabled, and the requests toward segment.io and wootric.com are still made.

<img width="1040" alt="Screenshot 2020-10-27 at 11 07 16" src="https://user-images.githubusercontent.com/953796/97287200-944f4d80-1844-11eb-811e-ccde72960721.png">

<img width="1230" alt="Screenshot 2020-10-27 at 11 06 55" src="https://user-images.githubusercontent.com/953796/97287183-8f8a9980-1844-11eb-89a4-5c10023edfa1.png">

**Comment by hazcod** [comment]
Can confirm for the latest preview release on Apple Silicon.

**Comment by MisterMike** [comment]
Same here, without any consent notification, and upon first-run of Docker, there are connections to wootric, segment.io and multiple connections to bugsnag. Thank god I have Little Snitch.

**Comment by shanejx** [comment]
This is still occurring on Mac OS Catalina. I just had Little Snitch warn me about connections to wootrics before I even could access the Preferences/Settings.

**Comment by c10l** [comment]
Still happening. 4 years and counting.

**Comment by pcjmfranken** [comment]
These connections to sessions.bugsnag.com, api.wootric.com, and api.segment.com despite an unmistakable opt-out through the "Send usage statistics" really need to stop. What you're doing here is probably illegal, but also a huge breach trust.

Your customers aren't your enemies.

**Comment by eastabrooka** [comment]
Update for Today : 
[***Featured on Hackernews*** ](https://news.ycombinator.com/item?id=34241473)

**Comment by RicardoMonteiroSimoes** [comment]
This is a big issue, doesn't this even fall, to some degree, under GRPC?

**Comment by RicardoMonteiroSimoes** [comment]
> Final remark: the consent should be opt-in [5](https://gdpr.eu/gdpr-consent-requirements/). So the checkbox should be disabled by-default. Can someone confirm that the checkbox is unchecked by default?

Can confirm that it was _not_ unchecked for me when I first installed it on my new macbook. It's turned on by default, and to be honest _I'm not sure but it might auto-turn-on after updates_, but I haven't kept an eye on it.

**Comment by Enrico204** [comment]
> > Final remark: the consent should be opt-in [5](https://gdpr.eu/gdpr-consent-requirements/). So the checkbox should be disabled by-default. Can someone confirm that the checkbox is unchecked by default?
> 
> Can confirm that it was _not_ unchecked for me when I first installed it on my new macbook. It's turned on by default, and to be honest _I'm not sure but it might auto-turn-on after updates_, but I haven't kept an eye on it.

That's not good.

If this flag is on by default, there is no consent because it has to be explicit (per GDPR). So, **if** there is any unique ID involved (for now, we're unsure) and PII (and, apparently, IP address is a PII, see my previous comment), I strongly suspect that this is not compliant with the GDPR.

However, I'm not a lawyer. I may be wrong.

**Comment by RicardoMonteiroSimoes** [comment]
bump because unacceptable, this might just push me to use an alternative to be honest

**Comment by c10l** [comment]
> bump because unacceptable, this might just push me to use an alternative to be honest

I've been using `colima` [0] on macOS for quite a while. It's great.

[0] https://github.com/abiosoft/colima

**Comment by MisterMike** [comment]
> > bump because unacceptable, this might just push me to use an alternative to be honest
> 
> I've been using `colima` [0] on macOS for quite a while. It's great.
> 
> [0] https://github.com/abiosoft/colima

Same here - more than happy (using with ddev)

**Comment by RicardoMonteiroSimoes** [comment]
Still no response.
